From 8a4cb593fba7ecacb4997952509451c6dfea9999 Mon Sep 17 00:00:00 2001
Message-Id: <8a4cb593fba7ecacb4997952509451c6dfea9999.1413837857.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: Durgadoss R <durgadoss.r@intel.com>
Date: Fri, 5 Sep 2014 14:32:39 +0530
Subject: [PATCH 310/312] FOR_UPSTREAM [VPG]: drm/i915: Fix the wrong punit
 read

Use vlv_cck_read to read fuse register instead of
punit_read.

We use bits[2:4] of this CCK_FUSE_REG to read and detect the platform SKU;
and accordingly obtain the cz clock and mem freq values. If we read P unit
registers (instead of CCK unit), then we may wrongly configure
these parameters.

Issue: GMINL-1043
Change-Id: I0a1585750ef0d26df4bd2b25b40e9dbebd223849
Signed-off-by: Durgadoss R <durgadoss.r@intel.com>
---
 drivers/gpu/drm/i915/intel_pm.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/intel_pm.c b/drivers/gpu/drm/i915/intel_pm.c
index e699dfe..624cf3b 100644
--- a/drivers/gpu/drm/i915/intel_pm.c
+++ b/drivers/gpu/drm/i915/intel_pm.c
@@ -6396,7 +6396,7 @@ static void cherryview_init_clock_gating(struct drm_device *dev)
 	u32 val;
 
 	mutex_lock(&dev_priv->rps.hw_lock);
-	val = vlv_punit_read(dev_priv, CCK_FUSE_REG);
+	val = vlv_cck_read(dev_priv, CCK_FUSE_REG);
 	mutex_unlock(&dev_priv->rps.hw_lock);
 	switch ((val >> 2) & 0x7) {
 	case 0:
-- 
1.7.9.5

